<a href="https://colab.research.google.com/github/alupo933/Project-2/blob/main/Project_2_Potential_Topics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd


In [3]:
filename = '/content/drive/MyDrive/Excel/wine_data.csv'
wine = pd.read_csv(filename)
wine.head()

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
0            7.4              0.70         0.00  ...       0.56      9.4        5
1            7.8              0.88         0.00  ...       0.68      9.8        5
2            7.8              0.76         0.04  ...       0.65      9.8        5
3           11.2              0.28         0.56  ...       0.58      9.8        6
4            7.4              0.70         0.00  ...       0.56      9.4        5

[5 rows x 12 columns]

In [4]:
wine.tail()

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
1594            6.2             0.600         0.08  ...       0.58     10.5        5
1595            5.9             0.550         0.10  ...       0.76     11.2        6
1596            6.3             0.510         0.13  ...       0.75     11.0        6
1597            5.9             0.645         0.12  ...       0.71     10.2        5
1598            6.0             0.310         0.47  ...       0.66     11.0        6

[5 rows x 12 columns]

First choice: dataset 1

1) Source of data
- Kaggle

2) Brief description of data
- This is a dataset of mutliple different red wines from around the world and their different attributes

3) What is the target?
- The target would be the quality based off of the attributes

4) Is this a classification or regression problem?
- This is a regression problem

5) How many features?
- 10

6) How many rows of data.
- 1598

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
- I don't forsee anu challenges other than correctly implementing bagging tree and random forests.



In [5]:
filename = '/content/drive/MyDrive/Excel/orthopedic data.csv'
ortho = pd.read_csv(filename)
ortho.head()

pelvic_incidence  pelvic_tilt numeric  ...  degree_spondylolisthesis     class
0         63.027818            22.552586  ...                 -0.254400  Abnormal
1         39.056951            10.060991  ...                  4.564259  Abnormal
2         68.832021            22.218482  ...                 -3.530317  Abnormal
3         69.297008            24.652878  ...                 11.211523  Abnormal
4         49.712859             9.652075  ...                  7.918501  Abnormal

[5 rows x 7 columns]

In [6]:
ortho.tail()

pelvic_incidence  pelvic_tilt numeric  ...  degree_spondylolisthesis   class
305         47.903565            13.616688  ...                 -4.245395  Normal
306         53.936748            20.721496  ...                 -0.421010  Normal
307         61.446597            22.694968  ...                 -2.707880  Normal
308         45.252792             8.693157  ...                  0.214750  Normal
309         33.841641             5.073991  ...                 -0.199249  Normal

[5 rows x 7 columns]

Second choice: dataset 2

1) Source of data
- Kaggle

2) Brief description of data
- This is a describption of different biomechanical features of orthopedics px. It describes the different ranges of motion tha they have at different joints in the px's body.

3) What is the target?
- The class of the orthopedic px's mobility (normal or abnormal)

4) Is this a classification or regression problem?
- This is a regression problem

5) How many features?
- 6

6) How many rows of data.
- 309

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
- There is limited data as there it is only on 309 patients, so the testing data may not be as accurate if it is adjested for overfitting.

In [7]:
filename = '/content/drive/MyDrive/Excel/pokemon_alopez247.csv'
pokemon = pd.read_csv(filename)
pokemon.head()

Number        Name Type_1  ... Weight_kg  Catch_Rate      Body_Style
0       1   Bulbasaur  Grass  ...       6.9          45       quadruped
1       2     Ivysaur  Grass  ...      13.0          45       quadruped
2       3    Venusaur  Grass  ...     100.0          45       quadruped
3       4  Charmander   Fire  ...       8.5          45  bipedal_tailed
4       5  Charmeleon   Fire  ...      19.0          45  bipedal_tailed

[5 rows x 23 columns]

In [8]:
pokemon.tail()

Number       Name   Type_1  ... Weight_kg  Catch_Rate       Body_Style
716     717    Yveltal     Dark  ...     203.0          45        two_wings
717     718    Zygarde   Dragon  ...     305.0           3  serpentine_body
718     719    Diancie     Rock  ...       8.8           3        head_arms
719     720      Hoopa  Psychic  ...       9.0           3        head_only
720     721  Volcanion     Fire  ...     195.0           3        quadruped

[5 rows x 23 columns]

Third choice: dataset 2

1) Source of data
- Kaggle

2) Brief description of data
- This is a dataset of all different pokemon and their different skill / physical attributes

3) What is the target?
- Type_1

4) Is this a classification or regression problem?
- This is a classification problem

5) How many features?
- 21

6) How many rows of data.
- 720 

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
- One of the challeges would be the multiclass classification conversions that would need to be done.